python/core/auto_generated/layertree/qgslayertreefilterproxymodel.sip.in

```diff
@@ -0,0 +1,113 @@
+/************************************************************************
+ * This file has been generated automatically from                      *
+ *                                                                      *
+ * src/core/layertree/qgslayertreefilterproxymodel.h                    *
+ *                                                                      *
+ * Do not edit manually ! Edit header and run scripts/sipify.pl again   *
+ ************************************************************************/
+
+
+
+
+
+class QgsLayerTreeFilterProxyModel : QSortFilterProxyModel
+{
+%Docstring
+QgsLayerTreeFilterProxyModel is a sort filter proxy model to easily reproduce the legend/layer tree in a tree view.
+Layers are checkable by default.
+Symbology nodes will not be shown.
+Layers can be fitlered by their type.
+
+For more complex use-cases, the model can be re-implemented to allow a different interaction or to add more columns.
+
+.. versionadded:: 3.14
+%End
+
+%TypeHeaderCode
+#include "qgslayertreefilterproxymodel.h"
+%End
+  public:
+    QgsLayerTreeFilterProxyModel( QObject *parent = 0 );
+%Docstring
+Constructor
+%End
+
+    void setCheckedLayers( const QList<QgsMapLayer *> layers );
+%Docstring
+Initialize the list of checked layers.
+
+.. note::
+
+   If the model is re-implemented, this method might become useless
+%End
+
+    QList<QgsMapLayer *> checkedLayers() const;
+%Docstring
+Returns the checked layers
+%End
+
+    QgsMapLayer *mapLayer( const QModelIndex &idx ) const;
+%Docstring
+Returns the map layer at a given index
+%End
+
+    QgsLayerTreeModel *layerTreeModel() const;
+%Docstring
+Rerturns the layer tree model
+%End
+    void setLayerTreeModel( QgsLayerTreeModel *layerTreeModel );
+%Docstring
+Sets the layer tree model
+%End
+
+    void resetLayerTreeModel();
+%Docstring
+This will refresh the model
+%End
+
+    void setMapLayerTypeFilter( const QList<QgsMapLayerType> &types = QList<QgsMapLayerType>() );
```

Comment:
Wouldn't it be better to reuse the filters from the map layer model proxy instead?

Reply:
good point, and done :)